### PR TITLE
ISPN-16699 RESP blocking during get with expiration in simple cache

### PR DESCRIPTION
**Pull request by jabolina**
https://issues.redhat.com/browse/ISPN-16699

### CODE REVIEW

File: core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java [hunk 1]
```diff
@@ -1090,7 +1092,7 @@ public CompletableFuture<Boolean> replaceAsync(K key, V oldValue, V newValue, Me
 
    @Override
    public CompletableFuture<V> getAsync(K key) {
-      return CompletableFuture.completedFuture(get(key));
+      return blockingManager.supplyBlocking(() -> get(key), "get-async").toCompletableFuture();
```
Comment (wburns) [inline core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java:1095]: I actually thought of a much cleaner way of doing this. Instead, we should use peek on the data container and then do the expiration check ourselves for the SimpleCacheImpl. This way we can return synchronously when not expired and if it is then we can submit the expiration call to the blocking manager.

File: core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java [hunk 2]
```diff
@@ -131,6 +132,7 @@ public class SimpleCacheImpl<K, V> implements AdvancedCache<K, V>, InternalCache
    @Inject CacheNotifier<K, V> cacheNotifier;
    @Inject TimeService timeService;
    @Inject KeyPartitioner keyPartitioner;
+   @Inject BlockingManager blockingManager;
```
Comment (wburns) [inline core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java:135]: Looks like we don't need this anymore.

**Review by wburns — COMMENTED** [review]
even better!

File: core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java (hunk 2 above)
Comment (jabolina) [reply core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java:135]: Oops!

**Review by wburns — APPROVED** [review]

**Comment by wburns** [comment]
Integrated into main, thanks @jabolina !

**Pull request merged**